wgpu-core/src/device/mod.rs

```diff
@@ -2610,7 +2610,15 @@ impl<A: HalApi> Device<A> {
                     {
                         break Some(pipeline::ColorStateError::FormatNotRenderable(cs.format));
                     }
-                    if cs.blend.is_some() && !format_features.flags.contains(Tfff::FILTERABLE) {
+                    if cs.blend.is_some()
+                        && (!format_features.flags.contains(Tfff::BLENDABLE)
+                            ||
+                            // according to WebGPU specifications the texture needs to be [`TextureFormatFeatureFlags::FILTERABLE`] 
+                            // if blending is set - use [`Features::TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES`] to elude this limitation
+                            (!self.features.contains(wgt::Features::TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES)
+                             && !format_features.flags.contains(Tfff::FILTERABLE)
+                            ))
+                    {
```

Comment:
This code is a bit hard to grok - could you split out individual booleans into named variables then combine them?

Reply:
Agreed. I've split them into their individual parts - should indeed help with the readability.